### ISSUE TITLE
BootstrapFewShot with max_rounds > 1 fails

### ISSUE THREAD

**Issue by kpob** [stanfordnlp/dspy#1612]
I started my dspy journey  with the 2.5.3 version. Following the quickstart example I created a llm instance
```python
dspy.OpenAI(...)
```
However this approach seems to be deprecated, so now I use 
```python
dspy.LM(...)
```

I was using the BootstrapFewShot for a while. It worked until I increased to max_round param to 2, from the default 1.

The error is
```
  File "/home/kpob/workspace/python/dspy-test/lib/python3.10/site-packages/dspy/teleprompt/bootstrap.py", line 182, in _bootstrap_one_example
    lm = lm.copy(temperature=0.7 + 0.001 * round_idx) if round_idx > 0 else lm
AttributeError: 'LM' object has no attribute 'copy'
```

My issue is similar to https://github.com/stanfordnlp/dspy/issues/1391. Yet I do not implement my custom LM, but just passing a model as a teacher.
```python
 def __init__(self, metric, lm: dspy.LM, program, program_name: str, metric_threshold=1.0):
        self.lm = lm
        self.metric = metric
        self.metric_threshold = metric_threshold
        self.program_name = program_name
        self.program = program

def __call__(self, trainset, record=True, experiment=""):
    optimizer = dspy.BootstrapFewShot(
            metric=self.metric, 
            metric_threshold=self.metric_threshold,
            max_labeled_demos=2,
            max_rounds=3,
            teacher_settings=dict(lm=self.lm)
    )
 ```

**Comment by okhat** [comment]
Hey, thanks @kpob for the catch. This is a new bug and I'll fix it today if I can.

**Comment by xiayouran** [comment]
@okhat @kpob I has same issue, I fix the bug #1621

**Comment by okhat** [comment]
Thanks @xiayouran ! Left some comments on the PR.

**Comment by okhat** [comment]
Fixed in 2.5.12
